### ISSUE TITLE
[Docs]: React - Proptypes

### ISSUE THREAD

**Issue by tmobaird** [tmobaird/i-want-to-contribute#80]
A Proptypes doc already exists for this project's doc book, but is currently an empty place holder until its contents are filled in.

This doc should be updated to include all things proptype related for react. Including things like:
- What is typechecking? 
- What are PropTypes?
- Why use PropTypes?
- How to use PropTypes

Also this doc should make sure we are referencing PropTypes usage from the recent React 15.5 or later. This is because this release moved `PropTypes` out of the main React node module and into a separate module of its own.

Code example should be used when possible! 🙂

**Comment by sherlyc** [comment]
Can I work on this?

**Comment by tmobaird** [comment]
@sherlyc definitely, it's all yours! Feel free to reach out to me if you have any other questions about helping out!
